hydra-node/src/Hydra/Ledger/Cardano.hs

```diff
@@ -141,22 +147,45 @@ instance Crypto crypto => ToJSONKey (Cardano.TxIn crypto) where
 instance Crypto crypto => ToJSON (Cardano.TxOut (MaryEra crypto)) where
   toJSON (Cardano.TxOut addr value) =
     object
-      -- TODO: Use ledger's instance, which is a base16-encoded
-      -- serialized address. We might want to use bech32
-      -- serialization in the end.
-      [ "address" .= addr
+      [ "address" .= serialiseAddressBech32
       , "value" .= valueToJson value
       ]
+   where
+    -- Serialise addresses in bech32 including the prefix as standardized:
+    -- https://github.com/cardano-foundation/CIPs/blob/master/CIP-0005/CIP-0005.md
+    serialiseAddressBech32 =
+      Bech32.encodeLenient prefix . Bech32.dataPartFromBytes $ Cardano.serialiseAddr addr
+
+    -- REVIEW(SN): The ledger's 'Addr' type is bigger than we need here and we
+    -- are forced to come up with a prefix for Byron "bootstrap" addresses,
+    -- although they should actually be serialised differently..and would not be
+    -- relevant for Hydra in the first place.
+    prefix = case addr of
+      (Cardano.Addr Cardano.Mainnet _ _) -> [Bech32.humanReadablePart|addr|]
```

Comment:
Is there no alternative to using Template haskell here?

Reply:
The alternative is `humanReadablePartFromText :: Text -> Maybe HumanReadablePart` + an `error "should not happen"` because we know that it won't fail for these 3 literals.

So even though I do personally dislike template haskell and quasi quotation, I think it's the better alternative to get compile time errors (or not) here.